pkg/metrics/tls.go

```diff
@@ -49,17 +61,30 @@ type TLSConfigurerImpl struct {
 	certDir           string
 	clientCAConfigMap string
 	clientCANamespace string
+	k8sWatcher        *k8scfgmap.ConfigMapWatcher
+
+	clientCAs       []*x509.Certificate
+	serverCerts     []tls.Certificate
+	tlsConfigHolder *certwatch.TLSConfigHolder
+
+	mutex sync.RWMutex
 }
 
 var _ TLSConfigurer = (*TLSConfigurerImpl)(nil)
 
 // NewTLSConfigurer creates a new TLS configurer.
-func NewTLSConfigurer(certDir string, clientCANamespace, clientCAConfigMap string) (TLSConfigurer, error) {
+func NewTLSConfigurer(certDir string, k8sClient kubernetes.Interface, clientCANamespace, clientCAConfigMap string) (TLSConfigurer, error) {
```

Comment:
Also no need to expose this one, since you already have `ConfigurerFromEnv` and the rest of the use-cases seem to be test related. Can always expose it if required I'd say.